tests/submodule/add.c

```diff
@@ -128,3 +128,69 @@ void test_submodule_add__url_relative_to_workdir(void)
 
 	assert_submodule_url("TestGitRepository", git_repository_workdir(g_repo));
 }
+
+void test_submodule_add__path_exists_in_index(void)
+{
+	git_index *index;
+	git_submodule *sm;
+	git_buf dirname = GIT_BUF_INIT;
+	git_buf filename = GIT_BUF_INIT;
+	FILE *fd;
+
+	/* In this repo, HEAD (master) has no remote tracking branc h*/
+	g_repo = cl_git_sandbox_init("testrepo");
+
+	git_buf_joinpath(&dirname, git_repository_workdir(g_repo), "TestGitRepository");
+	git_buf_joinpath(&filename, dirname.ptr, "test.txt");
+
+	p_mkdir(dirname.ptr, 0700);
+	fd = fopen(filename.ptr, "w");
+	fclose(fd);
+
+	cl_git_pass(
+		git_repository_index__weakptr(&index, g_repo)
+	);
```

Comment:
There's no need for the newlines here. In most cases, we simply ignore line length limits for our tests and put those things in one line.